src/PS5Parser.h

```diff
@@ -0,0 +1,392 @@
+/* Copyright (C) 2014 Kristian Lauszus, TKJ Electronics. All rights reserved.
+
+ This software may be distributed and modified under the terms of the GNU
+ General Public License version 2 (GPL2) as published by the Free Software
+ Foundation and appearing in the file GPL2.TXT included in the packaging of
+ this file. Please note that GPL2 Section 2[b] requires that all works based
+ on this software must also be made publicly available under the terms of
+ the GPL2 ("Copyleft").
+
+ Contact information
+ -------------------
+
+ Kristian Lauszus, TKJ Electronics
+ Web      :  http://www.tkjelectronics.com
+ e-mail   :  kristianl@tkjelectronics.com
+
+ PS5 support by Joseph Duchesne
+ */
+
+#ifndef _ps5parser_h_
+#define _ps5parser_h_
+
+#include "Usb.h"
+#include "controllerEnums.h"
+#include "PS5Trigger.h"
+
+/** Buttons on the controller */
+const uint8_t PS5_BUTTONS[] PROGMEM = {
+        UP, // UP
+        RIGHT, // RIGHT
+        DOWN, // DOWN
+        LEFT, // LEFT
+
+        0x0C, // SHARE
+        0x0D, // OPTIONS
+        0x0E, // L3
+        0x0F, // R3
+
+        0x0A, // L2
+        0x0B, // R2
+        0x08, // L1
+        0x09, // R1
+
+        0x07, // TRIANGLE
+        0x06, // CIRCLE
+        0x05, // CROSS
+        0x04, // SQUARE
+
+        0x10, // PS
+        0x11, // TOUCHPAD
+};
+
+union PS5Buttons {
+        struct {
+                uint8_t dpad : 4;
+                uint8_t square : 1;
+                uint8_t cross : 1;
+                uint8_t circle : 1;
+                uint8_t triangle : 1;
+
+                uint8_t l1 : 1;
+                uint8_t r1 : 1;
+                uint8_t l2 : 1;
+                uint8_t r2 : 1;
+                uint8_t share : 1;
+                uint8_t menu : 1;
+                uint8_t left_stick : 1;
+                uint8_t right_stick : 1;
+                
+
+                uint8_t ps : 1;
+                uint8_t touchpad : 1;
+                uint8_t mic : 1;
+                uint8_t dummy : 5;
+        } __attribute__((packed));
+        uint32_t val : 24;
+} __attribute__((packed));
+
+struct touchpadXY {
+        struct {
+                uint8_t counter : 7; // Increments every time a finger is touching the touchpad
+                uint8_t touching : 1; // The top bit is cleared if the finger is touching the touchpad
+                uint16_t x : 12;
+                uint16_t y : 12;
+        } __attribute__((packed)) finger[2]; // 0 = first finger, 1 = second finger
+} __attribute__((packed));
+
+struct PS5Status {
+        // first byte
+        uint8_t headphone : 1;
+        uint8_t dummy : 2;
+        uint8_t usb : 1;  // charging
+        uint8_t dummy2: 4;
+
+        // second byte
+        uint8_t battery : 4;
+        uint8_t dummy3 : 1;
+        uint8_t battery_full : 1;
+        uint8_t dummy4 : 2;
```

Comment:
I do not think this is correct. These are all 0's on my controller except for the 1st bit, which seems to be used for the microphone. I do not know about the `battery_full` bit though, as I'm not sure if my controller is fully charged or not.

Reply:
fixed

Reply:
@@josephduchesne what is your take on this?